internal/app/backend/backend_service.go

```diff
@@ -99,15 +99,15 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 		case <-stream.Context().Done():
 			return stream.Context().Err()
 		default:
-			err := stream.Send(&pb.FetchMatchesResponse{Match: result})
+			err = stream.Send(&pb.FetchMatchesResponse{Match: result})
+			telemetry.IncrementCounter(stream.Context(), mMatchesFetched)
 			if err != nil {
 				logger.WithError(err).Error("failed to stream back the response")
 				return err
 			}
 		}
 	}
 
-	telemetry.IncrementCounterN(stream.Context(), mMatchesFetched, len(results))
```

Comment:
Whats the root cause here? Why does IncrementCounterN here fail and IncrementCounter for each match added pass?

Reply:
Most of the emitted metrics are not recorded because function returns from the `stream.Context().Done()` case